### PR TITLE
1763: Refactor indicator builders to classes (2/3)

### PR DESCRIPTION
**Pull request by kael89**
### Issue #:

Towards beyondessential/tupaia-backlog#1763, Part 2/3

This is a pure refactoring that introduces a class interface to indicator builders. The reason for this is the need to associate more than one function with a specific indicator builder (eg `ArithmeticBuilder.buildAnalyticValues()` and `ArithmeticBuilder.getDataElementsToFetch()` => to be introduced in Part 3).

This PR is not as big as it seems! I recommend going through each commit one at a time, since a big majority of them are just moving things around.

### Commit description

1. [Use classes for indicator builders](https://github.com/beyondessential/tupaia/pull/1823/commits/249732c65f19d0aad777b399464f05381a15d3ee)
   * Minimum changes to introduce a class structure for indicators builders
2. [Make ArithmeticBuilder more classy](https://github.com/beyondessential/tupaia/pull/1823/commits/2522b30cdc6a471a9dd2b7f7b6165ff09f232c8e)
   * Add functions to the interface of the new Builder classes
3. [Nest test blocks](https://github.com/beyondessential/tupaia/pull/1823/commits/fb07d2d82a9349a127da1e50b367b7bc864f08a1)
    * Simply put all `describe blocks` under a new `buildAnalyticValues` describe block, no other changes
4. [Group related functionality in test blocks](https://github.com/beyondessential/tupaia/pull/1823/commits/624b414de1a6803cd60635ca22a4db19ff386ea8)
    * Simply group `describe` blocks under new `describe` blocks in `ArithmeticBuilder.test.ts`, no other changes
5. [Extract test blocks in files](https://github.com/beyondessential/tupaia/pull/1823/commits/80675d6cf1f3bf8248f32249a813be5584595f64)
   * Simply move the top level `describe` blocks to new files, no other changes

 The rest of commits are all small changes

### CODE REVIEW

**Review by kael89 — COMMENTED** [review]
In this commit:

* Minimum changes to introduce a class structure for indicators builders

**Review by rohan-bes — APPROVED** [review]
Looks good man!

**Comment by kael89** [comment]
Closing in favour of https://github.com/beyondessential/tupaia/pull/1869
